spec/features/permissions/updating_permission_with_restricted_collections_spec.rb

```diff
@@ -0,0 +1,285 @@
+# when there are selected collections that they do not have access to see. We need
+# to test and make sure that scenario has been prevented
+
+require 'rails_helper'
+
+describe 'Updating Collection Permissions when collections are not accessible by the user' do
+  # this collection should be visible to all Registered users
+  let(:entry_title_visible_to_all) { 'Near-Real-Time SSMIS EASE-Grid Daily Global Ice Concentration and Snow Extent V004' }
+  let(:entry_id_visible_to_all) { 'NISE_4' }
+
+  # these collections should only be visible to admin users
+  let(:restricted_entry_title_1) { 'MODIS/Aqua Sea Ice Extent and IST Daily L3 Global 4km EASE-Grid Day V005' }
+  let(:restricted_entry_id_1) { 'MYD29E1D_5' }
+  let(:restricted_entry_title_2) { 'AMSR-E/Aqua Daily L3 12.5 km Tb, Sea Ice Conc., & Snow Depth Polar Grids V003' }
+  let(:restricted_entry_id_2) { 'AE_SI12_3' }
+
+  before :all do
+    @collection_permission_some_restricted_name = "Testing Collection Permission with SOME restricted collections #{rand(1000)}"
+
+    collection_permission_some_restricted = {
+      group_permissions: [{
+        user_type: 'registered',
+        permissions: [ "read", "order" ]
+      }],
+      catalog_item_identity: {
+        "name": @collection_permission_some_restricted_name,
+        "provider_id": "NSIDC_ECS",
+        "collection_applicable": true,
+        "granule_applicable": true,
+        "collection_identifier": {
+          "entry_titles": [
+            "MODIS/Aqua Sea Ice Extent and IST Daily L3 Global 4km EASE-Grid Day V005",
+            "AMSR-E/Aqua Daily L3 12.5 km Tb, Sea Ice Conc., & Snow Depth Polar Grids V003",
+            "Near-Real-Time SSMIS EASE-Grid Daily Global Ice Concentration and Snow Extent V004"
+          ]
+        }
+      }
+    }
+
+    @collection_permission_some_restricted = cmr_client.add_group_permissions(collection_permission_some_restricted, 'access_token_admin').body
+
+    wait_for_cmr
+
+    @collection_permission_all_restricted_name = "Testing Collection Permission with ALL restricted collections #{rand(1000)}"
+
+    collection_permission_all_restricted = {
+      group_permissions: [{
+        user_type: 'registered',
+        permissions: [ "read", "order" ]
+      }],
+      catalog_item_identity: {
+        "name": @collection_permission_all_restricted_name,
+        "provider_id": "NSIDC_ECS",
+        "collection_applicable": true,
+        "granule_applicable": false,
+        "collection_identifier": {
+          "entry_titles": [
+            "MODIS/Aqua Sea Ice Extent and IST Daily L3 Global 4km EASE-Grid Day V005",
+            "AMSR-E/Aqua Daily L3 12.5 km Tb, Sea Ice Conc., & Snow Depth Polar Grids V003"
+          ]
+        }
+      }
+    }
+
+    @collection_permission_all_restricted = cmr_client.add_group_permissions(collection_permission_all_restricted, 'access_token_admin').body
+
+    wait_for_cmr
+  end
+
+  # need to delete?
```

Comment:
oops. I'll remove this. it was a placeholder for me to decide if I needed to delete the collection permissions after the tests ran.